File: VB/09_Interaction/FormField/AddTooltipForFormField.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Pdf
Imports Spire.Pdf.Graphics
Imports Spire.Pdf.Fields

Namespace AddTooltipForFormField
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'create a pdf document
			Dim doc As New PdfDocument()

			'add a new page
			Dim page As PdfPageBase = doc.Pages.Add(PdfPageSize.A4, New PdfMargins(0))

			'create a new pdf font
			Dim font As New PdfFont(PdfFontFamily.Helvetica, 12f, PdfFontStyle.Bold)

			'create a pdf brush
			Dim brush As PdfBrush = PdfBrushes.Black

			Dim x As Single = 50
			Dim y As Single = 50
			Dim tempX As Single = 0

			Dim text As String = "E-mail: "

			'draw a text into page
			page.Canvas.DrawString(text, font, brush, x, y)

			tempX = font.MeasureString(text).Width + x + 15

			'create a pdf textbox field
			Dim textbox As New PdfTextBoxField(page, "TextBox")

			'set the bounds of textbox field
			textbox.Bounds = New RectangleF(tempX, y, 100, 15)

			'set the border width of textbox field
			textbox.BorderWidth = 0.75f

			'set the border style of textbox field
			textbox.BorderStyle = PdfBorderStyle.Solid

			'add the textbox field into pdf document
			doc.Form.Fields.Add(textbox)

			'add a tooltip for the textbox field
			doc.Form.Fields("TextBox").ToolTip = "Please insert a valid email address"

			Dim output As String = "AddTooltipForFormField.pdf"

			'save pdf document
			doc.SaveToFile(output)

			'Launching the Pdf file
			PDFDocumentViewer(output)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/09_Interaction/FormField/AddTooltipForFormField.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Fields;

namespace AddTooltipForFormField
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //create a pdf document
            PdfDocument doc = new PdfDocument();

            //add a new page
            PdfPageBase page = doc.Pages.Add(PdfPageSize.A4, new PdfMargins(0));

            //create a new pdf font
            PdfFont font = new PdfFont(PdfFontFamily.Helvetica, 12f, PdfFontStyle.Bold);

            //create a pdf brush
            PdfBrush brush = PdfBrushes.Black;

            float x = 50;
            float y = 50;
            float tempX = 0;

            string text = "E-mail: ";
            
            //draw a text into page
            page.Canvas.DrawString(text, font, brush, x, y);

            tempX = font.MeasureString(text).Width + x + 15;

            //create a pdf textbox field
            PdfTextBoxField textbox = new PdfTextBoxField(page, "TextBox");

            //set the bounds of textbox field
            textbox.Bounds = new RectangleF(tempX, y, 100, 15);

            //set the border width of textbox field
            textbox.BorderWidth = 0.75f;

            //set the border style of textbox field
            textbox.BorderStyle = PdfBorderStyle.Solid;

            //add the textbox field into pdf document
            doc.Form.Fields.Add(textbox);

            //add a tooltip for the textbox field
            doc.Form.Fields["TextBox"].ToolTip = "Please insert a valid email address";

            string output = "AddTooltipForFormField.pdf";

            //save pdf document
            doc.SaveToFile(output);

            //Launching the Pdf file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
